ceph/rados/core_workflows.py

```diff
@@ -870,24 +876,46 @@ def change_osd_state(self, action: str, target: int) -> bool:
             target: ID osd the target OSD
         Returns: Pass -> True, Fail -> False
         """
-
         cluster_fsid = self.run_ceph_command(cmd="ceph fsid")["fsid"]
-        osd_nodes = self.ceph_cluster.get_nodes(role="osd")
-        flag = False
-        for onode in osd_nodes:
-            res = self.collect_osd_daemon_ids(osd_node=onode)
-            if target in res:
-                cmd = f"systemctl {action} ceph-{cluster_fsid}@osd.{target}.service"
-                flag = True
-                log.info(
-                    f"Performing {action} on osd-{target} on host {onode.hostname}. Command {cmd}"
-                )
-                onode.exec_command(sudo=True, cmd=cmd)
-                # Sleeping for 5 seconds for status to be updated
-                time.sleep(5)
-                return True
-        if not flag:
-            log.error(f"osd-{target} not found on cluster")
+        host = self.fetch_host_node(daemon_type="osd", daemon_id=str(target))
+        if not host:
+            log.error("failed to find host for the osd")
+            return False
+        cmd = f"systemctl {action} ceph-{cluster_fsid}@osd.{target}.service"
+        log.info(
+            f"Performing {action} on osd-{target} on host {host.hostname}. Command {cmd}"
+        )
+        host.exec_command(sudo=True, cmd=cmd)
+        # Sleeping for 5 seconds for status to be updated
+        time.sleep(5)
+        return True
+
+    def fetch_host_node(self, daemon_type: str, daemon_id: str):
+        """
+        Provides the Ceph cluster object for the given daemon. ceph_cluster
+        Args:
+            daemon_type: type of daemon
+                Allowed values: alertmanager, crash, mds, mgr, mon, osd, rgw, prometheus, grafana, node-exporter
+            daemon_id: name of the daemon, ID in case of OSD's
+
+        Returns: ceph object for the node
+
+        """
+        host_nodes = self.ceph_cluster.get_nodes()
+        all_daemons = self.run_ceph_command(cmd="ceph orch ps")
```

Comment:
We could change this to `ceph orch ps --daemon_type {daemon_type}`.. 
```
command = f"ceph orch ps --daemon_type {daemon_type}"
if daemon_id:
  command += " --daemon_id {daemon_id}"
daemons = self.run_ceph_command(cmd=command)
hosts = [ daemon["hostname"] for daemon in daemons]
return hosts
```

Reply:
made changes. moved the search of node to remote cluster